pkg/istio-agent/agent.go

```diff
@@ -351,20 +355,47 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 		return nil, fmt.Errorf("failed to start local DNS server: %v", err)
 	}
 
-	socketExists, err := checkSocket(ctx, security.WorkloadIdentitySocketPath)
+	// There are a couple of things we have to do here
+	//
+	// 1. Use a custom SDS workload socket if one is found+healthy at the configured path
+	// 3. Error out if a custom SDS socket path is configured but no socket is found there.
+	// 4. Do NOT error out, but just start and use the default Istio SDS server, if no socket
+	// is found AND no custom SDS socket path is configured.
+	//
+	//
+	// We do that like so
+	// 1. compare the (hardcoded) Istio default SDS server path with the (configurable) client path.
+	// 2. if they are different, do not start the Istio default SDS server.
+	// 3. if they are different, and no socket exists at that location, error out.
+	// 4. if they are not different, start the Istio default SDS server and use it.
+
+	// Correctness check - we do not want people sneaking paths into this
```

Comment:
I'm not sold on this. There isn't really a security violation - pod owners have total control over their pod. And I don't want to have to have yet another tweak to this code path in 6 months when someone needs a custom path because we were too opinionated. WDYT?

Reply:
I swear there was a conversation about this where this was brought up as a sticking point for doing this previously, but I can't remember/find it.

> And I don't want to have to have yet another tweak to this code path in 6 months when someone needs a custom path because we were too opinionated. WDYT?

That occurred to me. I am slightly in favor of having it be fixed/opinionated here tho, for the following reasons.

- All the other paths in `security.go` are already similarly hardcoded/nonconfigurable, including the SDS ones. It's your pod yes, but our mount, and our sidecar.
- We document and support and recommend doing this in exactly one way - via the SPIRE CSI driver. And if you use that, you by-definition never need to alter this path - making the whole path configurable just makes it easier to not-follow our docs and do something that is less secure and operationally harmful that we do not support or recommend (raw hostmounts, no CSI driver). I literally want to make it hard to do something that both SPIRE and us explicitly recommend against and (in our case) do not currently document or support.


basically, there is no valid case for using an alt path other than to avoid the SPIRE CSI driver, which is a deployment model we explicitly do not recommend or document.

Reply:
alright fine with me, I'm convinced. One nice thing is we can keep the same API and allow full path later if needed